src/useScreenSize.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { FC, useEffect } from 'react'
+
+export interface ScreenSize {
+  width: number
+  height: number
+  isMobile: boolean
+}
+
+function useScreenSize() {
+  const [{ width, height, isMobile }, setSize] = React.useState<ScreenSize>({
+    width: 0,
+    height: 0,
+    isMobile: true
+  })
+
+  useEffect(() => {
+    if (typeof window === 'undefined') return
+
+    window.addEventListener('resize', handleResize)
+
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
+
+  function handleResize() {
+    setSize({
+      width: window.innerWidth,
+      height: window.innerHeight,
+      isMobile: window.innerWidth < 768
+    })
+  }
```

Comment:
I have tried it, and found a bug.

Set your viewport to a large screen size and hit **Refresh.** It shows the small version instead. I have to resize the window once before it shows the large version.

Maybe you can create a function `const getSize = () => ({ width: …, … })` and use it in `useState` initializer and `handleResize`.

This project does not do any server-side rendering so it is safe to access the viewport on first render.

Reply:
I 

> I have tried it, and found a bug.
> 
> Set your viewport to a large screen size and hit **Refresh.** It shows the small version instead. I have to resize the window once before it shows the large version.
> 
> Maybe you can create a function `const getSize = () => ({ width: …, … })` and use it in `useState` initializer and `handleResize`.
> 
> This project does not do any server-side rendering so it is safe to access the viewport on first render.

I will try krub 😉